drivers/display/display_ili9xxx.c

```diff
@@ -250,14 +250,24 @@ static int ili9xxx_set_orientation(const struct device *dev,
 	uint8_t tx_data = ILI9XXX_MADCTL_BGR;
 
 	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+#if !CONFIG_ILI9342C
```

Comment:
Commit message does not explain why it is necessary. Anyway, right way could be
```c

	if (orientation == DISPLAY_ORIENTATION_NORMAL &&
	    !IS_ENABLED(CONFIG_ILI9342C)) {

```

But it breaks support of multiple instances. Maybe specific compatibles can be a solution, see https://github.com/zephyrproject-rtos/zephyr/pull/48163

Reply:
I didn't get the solution exactly, i can see for ili9xxx series there are specific compatibles per controller model like:
`ilitek,ili9340`, `ilitek,ili9341`, `ilitek,ili9488`.
The problem is this model 9342c has different Orientation settings than others, as an example [See here](https://github.com/letscontrolit/ESPEasy/blob/5f40dafc5ac315ea4ddd623531d71de33218506d/lib/Adafruit_ILI9341/Adafruit_ILI9341.cpp#L623)

What do you think if an orientation function implemented per model as part of [config struct](https://github.com/zephyrproject-rtos/zephyr/blob/fb60aab2458684f347533cba3658c89757580932/drivers/display/display_ili9xxx.h#L57) ? then that function will be called by `ili9xxx_set_orientation` ?

Reply:
The first step should be to identify the reason for the differences, MADCTL(0x36) register description looks the same for example for ILI9340 and ILI9342C. You can then use DT_HAS_COMPAT_STATUS_OKAY(ilitek,ili9342c) to add a quirk to specific driver instance or just set a boolean in the driver's config.

Reply:
I tried to identify why this controller behave differently but couldn't find a clear answer. 
According to the datasheet it should be the same MADCTL command used in the other models, but dunno why it behaves differently, checked other drivers on Github also, all they do is handling this controller differently.
Also, in the datasheet the part about init code is not in 9342c datasheet.
So I think the solution would be to add a quirk to the driver like the example you mentioned above.

Reply:
The mystery of different behavior of MADCTL in this specific model not solved yet, I even asked m5stack and they don't have an answer.
So, what is the Zephyr way here to tell the user orientation is not support message?